[docs][DOC-656] CDC: Add content for parallel streaming support using logical replication (#26254)
yugabyte/yugabyte-db: docs/content/preview/develop/change-data-capture/using-logical-replication/yugabytedb-connector-properties.md

* SEQUENCE - A monotonic increasing number that determines the record in global order within the context of a slot.
* HYBRID_TIME - A hybrid time value that can be used to compare transactions across slots.

##### streaming.mode

Specified whether the connector should use a single task mode to stream changes or multi task mode.

* `default` uses the single task mode and streams all the changes using it.

> * `parallel` uses a multi task mode and streams changes using the number of specified replication slots.

Reviewer: You specify the slots to use using the slot.names, publication.names, or slot.ranges property.

{{< note title="Important" >}}

When deploying the connector with `parallel` streaming mode, the user will need to ensure that the `table.include.list` only contains one table for which the streaming is supposed to happen in parallel.

{{< /note >}}

{{< note title="Usage with snapshot" >}}

If `snapshot.mode` is set to `initial` or `initial_only`, the user also needs to ensure that the configuration also contains a valid value for the configuration property `primary.key.hash.columns`.

{{< /note >}}

##### slot.names

> A list of comma separated values of different slot names to be used by each task when using the `streaming.mode=parallel`.

Reviewer: ```suggestion
A list of slot names, provided as comma-separated values, to be used by each task when using `streaming.mode=parallel`.
```

No default.

> {{< warning title="Warning" >}}

Reviewer: I would remove the warning block, doesn't seem dangerous

> The configuration `slot.names` is only supposed to be used when `streaming.mode` is set to `parallel`, it won't have any effect otherwise.

Reviewer: ```suggestion
`slot.names` applies only when `streaming.mode` is set to `parallel`; otherwise it has no effect.
```

{{< /warning >}}

##### publication.names

> A list of comma separated values of different publication names to be used by each task when using the `streaming.mode=parallel`.

Reviewer: ```suggestion
A list of publication names, provided as comma-separated values, to be used by each task when using `streaming.mode=parallel`.
```

No default.

{{< warning title="Warning" >}}

> The configuration `publication.names` is only supposed to be used when `streaming.mode` is set to `parallel`, it won't have any effect otherwise.

Reviewer: same change and ditch the warning

{{< /warn >}}

> ##### publication.names

Reviewer 2: `publication.names` is already mentioned above.

> A list of semi-colon separated values of different hash code ranges to be used by each task when using the `streaming.mode=parallel`.

Reviewer: ```suggestion
A range of slots to be used by each task when using `streaming.mode=parallel`, provided as tablet hash code ranges separated by semi colons.
```

No default.

{{< warning title="Warning" >}}

> The configuration `slot.ranges` is only supposed to be used when `streaming.mode` is set to `parallel`, it won't have any effect otherwise.

Reviewer: same change and ditch the warning

{{< /warning >}}

> For example, suppose we have a table with 3 tablets where the tablets have hash ranges as `[0,21845)`, `[21845,43690)` and `[43690,65536)` then the value for this configuration would be `slot.ranges=0,21845;21845,43690;43690,65536`.

Reviewer: ```suggestion
For example, suppose you have a table with 3 tablets where the tablets have hash ranges of `[0,21845)`, `[21845,43690)`, and `[43690,65536)`. The value for this configuration would be `slot.ranges=0,21845;21845,43690;43690,65536`.
```

##### primary.key.hash.columns

The columns of the table which consitute the hash part of the primary key.

> {{< note title="Use only with parallel streaming mode" >}}

Reviewer: remove the note block

> This configuration is only supposed to be used when `streaming.mode` is set to `parallel`.

Reviewer: ```suggestion
This property is only valid when `streaming.mode` is set to `parallel`.
```
